src/main/java/uk/gov/companieshouse/registeredemailaddressapi/service/EligibilityService.java

```diff
@@ -0,0 +1,55 @@
+package uk.gov.companieshouse.registeredemailaddressapi.service;
+
+import java.util.List;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.stereotype.Service;
+
+import uk.gov.companieshouse.api.model.company.CompanyProfileApi;
+import uk.gov.companieshouse.registeredemailaddressapi.eligibility.EligibilityRule;
+import uk.gov.companieshouse.registeredemailaddressapi.eligibility.EligibilityStatusCode;
+import uk.gov.companieshouse.registeredemailaddressapi.exception.CompanyNotFoundException;
+import uk.gov.companieshouse.registeredemailaddressapi.exception.EligibilityException;
+import uk.gov.companieshouse.registeredemailaddressapi.exception.ServiceException;
+import uk.gov.companieshouse.registeredemailaddressapi.model.response.CompanyValidationResponse;
+import uk.gov.companieshouse.registeredemailaddressapi.utils.ApiLogger;
+
+@Service
+public class EligibilityService {
+
+    private final List<EligibilityRule<CompanyProfileApi>> eligibilityRules;
+
+    private final CompanyProfileService companyProfileService;
+
+    @Autowired
+    public EligibilityService(@Qualifier("rea-update-eligibility-rules") List<EligibilityRule<CompanyProfileApi>> eligibilityRules, CompanyProfileService companyProfileService){
+        this.eligibilityRules = eligibilityRules;
+        this.companyProfileService = companyProfileService;
+    }
+
+    public boolean checkCompanyEligibility(String companyNumber) throws ServiceException, CompanyNotFoundException, EligibilityException {
+        var companyProfile = companyProfileService.getCompanyProfile(companyNumber);
+
+        for (EligibilityRule<CompanyProfileApi> eligibilityRule : eligibilityRules) {
```

Comment:
Do we need a try catch eligibility exception here?

Reply:
No because the mechanism is to catch in the GlobalExceptionHandler (where is relevant return / status code is selected, in this case 403 forbidden)